server/anonymous/verified_plugins.go

```diff
@@ -0,0 +1,70 @@
+package anonymous
+
+import (
+	"fmt"
+	"time"
+
+	"github.com/mattermost/mattermost-server/v5/mlog"
+	"github.com/pkg/errors"
+)
+
+// PluginIdentifier unique plugin identifier
+type PluginIdentifier struct {
+	ID      string
+	Version string
+}
+
+// UnverifiedPlugins returns list of unverified plugins
+func (a *anonymous) UnverifiedPlugins() []PluginIdentifier {
+	a.unverifiedPluginsLock.RLock()
+	defer a.unverifiedPluginsLock.RUnlock()
+
+	return a.unverifiedPluginsList
+}
+
+// StartPluginChecks starts checking unverified plugins
+func (a *anonymous) StartPluginChecks() {
+	go func() {
+		for now := range time.Tick(time.Hour) {
+			mlog.Info(fmt.Sprintf("started updating validated plugins %s", now.String()))
+
+			plugins, err := a.unverifiedPlugins()
+			if err != nil {
+				mlog.Error(err.Error())
+				return
+			}
+
+			a.unverifiedPluginsLock.Lock()
+			a.unverifiedPluginsList = plugins
+			a.unverifiedPluginsLock.Unlock()
+		}
+	}()
+}
+
+func (a *anonymous) unverifiedPlugins() ([]PluginIdentifier, error) {
+
```

Comment:
Change the name of this function so it doesn't coincide with the name of the exported function that has different logic to it (change it to getUnverifiedPlugins for example)